File: Scripts/server/file-input-output/c/file-input-output.c
#include <stdio.h>


int executeTask(int i, FILE *in, FILE *out, int c){
  in = fopen("../input.txt", "r");
  if (!in) {
    fprintf(stderr, "Error opening input.txt for reading.\n");
    return 1;
  }

  out = fopen("output.txt", "w");
  if (!out) {
    fprintf(stderr, "Error opening output.txt for writing.\n");
    fclose(in);
    return 1;
  }

  while ((c = fgetc(in)) != EOF) {
    fputc(c, out);
  }

  fclose(out);
  fclose(in);

  return i;
}

int main(int argc, char **argv) {
  volatile FILE *in, *out;
  volatile int c;
  volatile int r;

for ( int i = 0; i < 1000000; ++i) {
	r = executeTask(i, &in, &out, c);
}	
  return 0;
}

File: Scripts/server/file-input-output/go/file-input-output.go
package main

import (
    "fmt"
    "io/ioutil"
)

func executeTask(i int) int {
    b, err := ioutil.ReadFile("../input.txt")
    if err != nil {
        fmt.Println(err)
       return 1
    }
    if err = ioutil.WriteFile("output.txt", b, 0666); err != nil {
        fmt.Println(err)
    }
	return i + 1
}

func main() {
r := int(1)
for i := 0; i < 1000000; i++ {
	r = executeTask(i + r) 
}
}
